gapic/schema/wrappers.py

```diff
@@ -1199,6 +1266,29 @@ def gen_indirect_resources_used(message):
             )
         )
 
+    @utils.cached_property
+    def resource_messages_dict(self) -> Dict[str, MessageType]:
+        """Returns a dict from resource reference to
+        the message type. This *includes* the common resource messages.
+
+        Returns:
+            Dict[str, MessageType]: A mapping from resource path
+                string to the corresponding MessageType.
+                `{"locations.googleapis.com/Location": MessageType(...)}`
+        """
+        service_resource_messages = dict(
+            (r.resource_type_full_path, r)
+            for r in list(self.resource_messages)
+        )
```

Comment:
Outrageous nit: slightly more elegant to do
```python
service_resource_messages = {r.resource_type_full_path : r for r in self.resource_messages}
```
Don't need to turn `resource_messages` into a list because it's already iterable.